### PR TITLE
KAFKA-5404: Add more AdminClient checks to ClientCompatibilityTest

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

**Comment by ijuma** [comment]
Can you please fix checkstyle errors? It would be good to run checkstyle locally before opening the PR. :)

File: tests/kafkatest/tests/client/client_compatibility_features_test.py [hunk 1]
```diff
@@ -23,18 +23,26 @@
...
 from ducktape.tests.test import Test
-from kafkatest.version import DEV_BRANCH, LATEST_0_10_0, LATEST_0_10_1, LATEST_0_10_2, V_0_10_1_0, KafkaVersion
+from kafkatest.version import DEV_BRANCH, LATEST_0_10_0, LATEST_0_10_1, LATEST_0_10_2, V_0_11_0_0, V_0_10_1_0, KafkaVersion
 
 def get_broker_features(broker_version):
     features = {}
     if (broker_version < V_0_10_1_0):
+        features["create-topics-supported"] = False
+    else:
+        features["create-topics-supported"] = True
```
Comment (ijuma) [inline tests/kafkatest/tests/client/client_compatibility_features_test.py]: Why are these in a separate block when compared to `offsets-for-times-supported`, etc.

File: tests/kafkatest/tests/client/client_compatibility_features_test.py [hunk 2]
```diff
@@ -74,17 +82,13 @@ def invoke_compatibility_program(self, features):
...
                "--num-cluster-nodes %d "
                "--topic %s " % (self.zk.path.script("kafka-run-class.sh", node),
                                self.kafka.bootstrap_servers(),
-                               features["offsets-for-times-supported"],
-                               features["cluster-id-supported"],
-                               features["expect-record-too-large-exception"],
                                len(self.kafka.nodes),
                                self.topics.keys()[0]))
+        for k, v in features.iteritems():
+            cmd = cmd + ("--%s %s " % (k, v))
```
Comment (ijuma) [inline tests/kafkatest/tests/client/client_compatibility_features_test.py:89]: Nice.

File: tools/src/main/java/org/apache/kafka/tools/ClientCompatibilityTest.java [hunk 3]
```diff
@@ -234,6 +260,68 @@ void testAdminClient() throws Exception {
...
+                        } catch (ExecutionException e) {
+                            if (e.getCause() instanceof SecurityDisabledException)
+                                return;
+                            throw e.getCause();
+                        }
+                    }
+                },
+                new ResultTester() {
+                    @Override
+                    public void test() throws Throwable {
```
Comment (ijuma) [inline tools/src/main/java/org/apache/kafka/tools/ClientCompatibilityTest.java:276]: Is it intentional that this is blank?

**Review by ijuma — COMMENTED** [review]
Thanks for the PR, great to have these tests. Looks good overall, left a few minor comments.

File: tests/kafkatest/tests/client/client_compatibility_features_test.py (hunk 2 above)
Comment (cmccabe) [reply tests/kafkatest/tests/client/client_compatibility_features_test.py:89]: yeah, it was getting annoying to add each thing manually.

File: tools/src/main/java/org/apache/kafka/tools/ClientCompatibilityTest.java (hunk 3 above)
Comment (cmccabe) [reply tools/src/main/java/org/apache/kafka/tools/ClientCompatibilityTest.java:276]: yes.  let me make it clearer

**Review by ijuma — APPROVED** [review]
Thanks for the PR, LGTM. Running it in the branch builder:

https://jenkins.confluent.io/job/system-test-kafka-branch-builder/914/

**Comment by ijuma** [comment]
Branch builder passed, merging to trunk and 0.11.0.

**Pull request closed without merging**